### PR TITLE
chore: regen api interface and address vulnerabilities

### PR DESCRIPTION
**Pull request by kleyow**

### CODE REVIEW

**Review by elnyry-sam-k — DISMISSED** [review]

File: src/server/handlers/consentRequests.ts [hunk 1]
```diff
@@ -68,12 +82,13 @@ async function post(_context: unknown, request: Request, h: ResponseToolkit): Pr
       Enum.Http.RestMethods.POST,
       payload,
       span
-    )
-    .catch((err: any) => {
-        // Do nothing with the error - forwardConsentRequestsRequest takes care of async errors
-        Logger.error('ConsentRequests::post - forwardConsentRequestsRequest async handler threw an unhandled error')
-        Logger.error(ReformatFSPIOPError(err))
-      })
+    ).catch((err) => {
```
Comment (mdebarros) [inline src/server/handlers/consentRequests.ts:85]:
```suggestion
    ).catch(err => {
```

File: src/server/handlers/consentRequests/{ID}.ts [hunk 2]
```diff
@@ -70,12 +83,13 @@ async function put(_context: unknown, request: Request, h: ResponseToolkit): Pro
       Enum.Http.RestMethods.PUT,
       payload,
       span
-    )
-    .catch(err => {
-        // Do nothing with the error - forwardConsentRequestsIdRequest takes care of async errors
-        Logger.error('ConsentRequests::put - forwardConsentRequestsIdRequest async handler threw an unhandled error')
-        Logger.error(ReformatFSPIOPError(err))
-      })
+    ).catch((err) => {
```
Comment (mdebarros) [inline src/server/handlers/consentRequests/{ID}.ts:86]:
```suggestion
    ).catch(err => {
```

File: src/server/handlers/consentRequests/{ID}.ts [hunk 3]
```diff
@@ -119,12 +141,13 @@ async function patch(_context: unknown, request: Request, h: ResponseToolkit): P
       Enum.Http.RestMethods.PATCH,
       payload,
       span
-    )
-    .catch(err => {
-        // Do nothing with the error - forwardConsentRequestsIdRequest takes care of async errors
-        Logger.error('ConsentRequests::put - forwardConsentRequestsIdRequest async handler threw an unhandled error')
-        Logger.error(ReformatFSPIOPError(err))
-      })
+    ).catch((err) => {
```
Comment (mdebarros) [inline src/server/handlers/consentRequests/{ID}.ts:144]:
```suggestion
    ).catch(err => {
```

File: src/server/handlers/consentRequests/{ID}/error.ts [hunk 4]
```diff
@@ -1,86 +1,101 @@
...
-         Logger.error(ReformatFSPIOPError(err))
-       })
+    // Note: calling async function without `await`
+    forwardConsentRequestsIdRequestError(
+      Enum.EndPoints.FspEndpointTemplates.TP_CONSENT_REQUEST_PUT_ERROR,
+      consentRequestsId,
+      request.headers,
+      payload,
+      span
+    ).catch((err) => {
```
Comment (mdebarros) [inline src/server/handlers/consentRequests/{ID}/error.ts:83]:
```suggestion
    ).catch(err => {
```

File: src/server/handlers/consents.ts [hunk 5]
```diff
@@ -72,10 +85,11 @@ async function post (_context: unknown, request: Request, h: ResponseToolkit): P
       Enum.Http.RestMethods.POST,
       payload,
       span
-    )
-    .catch((err: any) => {
+    ).catch((err) => {
```
Comment (mdebarros) [inline src/server/handlers/consents.ts:88]:
```suggestion
    ).catch(err => {
```

File: src/server/handlers/consents/{ID}.ts [hunk 6]
```diff
@@ -70,12 +84,13 @@ async function put(_context: unknown, request: Request, h: ResponseToolkit): Pro
       Enum.Http.RestMethods.PUT,
       payload,
       span
-    )
-    .catch(err => {
-        // Do nothing with the error - forwardConsentsIdRequest takes care of async errors
-        Logger.error('Consents::put - forwardConsentsIdRequest async handler threw an unhandled error')
-        Logger.error(ReformatFSPIOPError(err))
-      })
+    ).catch((err) => {
```
Comment (mdebarros) [inline src/server/handlers/consents/{ID}.ts:87]:
```suggestion
    ).catch(err => {
```

File: src/server/handlers/accounts/{ID}.ts [hunk 7]
```diff
@@ -119,12 +142,13 @@ const put = async (_context: unknown, request: Request, h: ResponseToolkit): Pro
       userId,
       payload,
       span
-    )
-      .catch(err => {
-        // Do nothing with the error - forwardAccountsIdRequest takes care of async errors
-        Logger.error('Accounts::put - forwardAccountsIdRequest async handler threw an unhandled error')
-        Logger.error(ReformatFSPIOPError(err))
-      })
+    ).catch((err) => {
```
Comment (mdebarros) [inline src/server/handlers/accounts/{ID}.ts:145]:
```suggestion
    ).catch(err => {
```

File: src/server/handlers/accounts/{ID}/error.ts [hunk 8]
```diff
@@ -66,12 +80,13 @@ const put = async (_context: unknown, request: Request, h: ResponseToolkit): Pro
       userId,
       payload,
       span
-    )
-      .catch(err => {
-        // Do nothing with the error - forwardAccountsIdRequestError takes care of async errors
-        Logger.error('Accounts::put:error - forwardAccountsIdRequestError async handler threw an unhandled error')
-        Logger.error(ReformatFSPIOPError(err))
-      })
+    ).catch((err) => {
```
Comment (mdebarros) [inline src/server/handlers/accounts/{ID}/error.ts:83]:
```suggestion
    ).catch(err => {
```

File: src/server/handlers/services/{ServiceType}.ts [hunk 9]
```diff
@@ -89,13 +104,13 @@ const get = async (_context: unknown, request: Request, h: ResponseToolkit): Pro
...
         payload,
         span
-      )
-        .catch(err => {
-          // Do nothing with the error - forwardGetServicesServiceTypeRequestFromProviderService takes care of async errors
-          Logger.error('Services::put - forwardGetServicesServiceTypeRequestFromProviderService async handler threw an unhandled error')
-          Logger.error(ReformatFSPIOPError(err))
-        })
-
+      ).catch((err) => {
```
Comment (mdebarros) [inline src/server/handlers/services/{ServiceType}.ts:107]:
```suggestion
      ).catch(err => {
```

Ok, there are a lot of these. This is the last one I am noting, but this comment is applicable everywhere else this appears.

I am surprised the linter was ok with this.

**Review by mdebarros — COMMENTED** [review]
No big issues for me, just the one suggested.

I do not feel particularly strongly about it, but I am surprised the linter allowed it?

**Review by mdebarros — APPROVED** [review]
+1

Lint/format issues to be fixed as part of https://github.com/mojaloop/project/issues/2731.

**Pull request merged**